MapboxMobileEvents/MMENSURLSessionWrapper.m

```diff
@@ -26,6 +34,10 @@ - (void)reconfigure:(MMEEventsConfiguration *)configuration {
     [self.certPin updateWithConfiguration:configuration];
 }
 
+-(void)dealloc {
+    [self.session invalidateAndCancel];
```

Comment:
I'm not sure this is going to fire @rclee @alfwatt: since `sessionWithConfiguration:delegate:delegateQueue:` retains self, `dealloc` won't fire. 

We'll need to call `invalidateAndCancel` somewhere earlier.

Reply:
Yup. Good catch.

Reply:
So, I'm a bit concerned with calling `invalidateAndCancel` earlier or at all. It doesn't seem like we should be recreating the session at all for the life of the app ([referencing "The Eskimo"](https://forums.developer.apple.com/thread/14785)). 

We really just need to avoid having multiple `NSURLSession`/`NSURLSessionWrapper`s in play. I think that may be what's causing our troubles in the first place.